### PR TITLE
Remove unneeded flag enable_randomness_ptb_restrictions

### PR DESCRIPTION
**Pull request by benr-ml**
## Description 

`enable_randomness_ptb_restrictions` is not needed since Random does not yet exist on testnet/mainnet.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/transaction.rs [hunk 1]
```diff
@@ -984,25 +984,22 @@ impl ProgrammableTransaction {
         }
 
         // A command that uses Random can only be followed by TransferObjects or MergeCoins.
-        if config.enable_randomness_ptb_restrictions() {
-            // Check if there is a random object in the input objects
```
Comment (lxfind) [inline crates/sui-types/src/transaction.rs:988]: Seems the comment was removed

File: crates/sui-framework/docs/deepbook/dependencies/sui-framework/hmac.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+
```
Comment (lxfind) [inline crates/sui-framework/docs/deepbook/dependencies/sui-framework/hmac.md:1]: What is this file?

File: crates/sui-types/src/transaction.rs (hunk 1 above)
Comment (benr-ml) [reply crates/sui-types/src/transaction.rs:988]: yes, I think it doesn't help the reader

File: crates/sui-framework/docs/deepbook/dependencies/sui-framework/hmac.md (hunk 2 above)
Comment (benr-ml) [reply crates/sui-framework/docs/deepbook/dependencies/sui-framework/hmac.md]: removed

**Review by lxfind — APPROVED** [review]

**Pull request merged**